tsfc/fem.py

```diff
@@ -340,15 +339,16 @@ def callback(key):
         table = ctx.tabulation_manager[key]
         if len(table.shape) == 1:
             # Cellwise constant
-            row = gem.Literal(table)
-            if numpy.count_nonzero(table) <= 2:
+            row = table
+            if numpy.count_nonzero(table.array) <= 2:
                 assert row.shape == vec.shape
                 return reduce(gem.Sum,
                               [gem.Product(gem.Indexed(row, (i,)), gem.Indexed(vec, (i,)))
                                for i in range(row.shape[0])],
                               gem.Zero())
         else:
-            table = ctx.index_selector(lambda i: gem.Literal(table[i]), mt.restriction)
+            table = ctx.index_selector(lambda i: gem.partial_indexed(table, (i,)),
```

Comment:
To some extent this is just code generation cosmetics. But `gem.Literal(table.array[i])` would produce nicer code I think than `gem.partial_indexed(table, (i,))`.